src/uucore/src/lib/lib.rs

```diff
@@ -84,3 +168,83 @@ pub fn args() -> impl Iterator<Item = String> {
 pub fn args_os() -> impl Iterator<Item = OsString> {
     wild::args_os()
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::ffi::OsStr;
+
+    fn make_os_vec(os_str: &OsStr) -> Vec<OsString> {
+        vec![
+            OsString::from("test"),
+            OsString::from("สวัสดี"),
+            os_str.to_os_string(),
+        ]
+    }
+
+    fn collect_os_str(vec: Vec<OsString>, handling: InvalidEncodingHandling) -> ConversionResult {
+        vec.into_iter().collect_str(handling)
+    }
+
+    fn test_invalid_utf8_args_lossy(os_str: &OsStr) {
+        //assert our string is invalid utf8
+        assert!(os_str.to_os_string().into_string().is_err());
+        let test_vec = make_os_vec(os_str);
+        let collected_to_str =
+            collect_os_str(test_vec.clone(), InvalidEncodingHandling::ConvertLossy)
+                .expect_lossy("Lossy conversion expected in this test: bad encoding entries should be converted as good as possible");
+        //conservation of length - when accepting lossy conversion no arguments may be dropped
+        assert_eq!(collected_to_str.len(), test_vec.len());
+        //first indices identical
+        for index in 0..2 {
+            assert_eq!(
+                collected_to_str.get(index).unwrap(),
+                test_vec.get(index).unwrap().to_str().unwrap()
+            );
+        }
+        //lossy conversion for string with illegal encoding is done
+        assert_eq!(
+            *collected_to_str.get(2).unwrap(),
+            os_str.to_os_string().to_string_lossy()
+        );
+    }
+
+    fn test_invalid_utf8_args_ignore(os_str: &OsStr) {
```

Comment:
ditto?

Reply:
Same here: Called from line 248 in lib.rs allows for testing broken encoding on different platforms.